### ISSUE TITLE
/api/v1/report/accounts/ is not returning the correct number of instances

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#475]
In GitLab by @erezende on Aug 23, 2018, 16:15

## Summary

Considering that there are two accounts and the usage for the last 30 days is as the following screenshots:

![Screenshot_from_2018-08-23_17-08-09](/uploads/caaed3c1338dc68192891441f96ecde8/Screenshot_from_2018-08-23_17-08-09.png)

![Screenshot_from_2018-08-23_17-08-30](/uploads/1a5de4e5f7bb9149a19116064207021d/Screenshot_from_2018-08-23_17-08-30.png)

The account summary report is not returning the right numbers:

```
{'cloud_account_overviews': [{'id': 1,
   'cloud_account_id': '885089032249',
   'user_id': 2,
   'type': 'aws',
   'arn': 'arn:aws:iam::885089032249:role/mock-arn',
   'creation_date': '2017-08-23T18:43:46.784643Z',
   'name': 'My account',
   'images': 3,
   'instances': 7,
   'rhel_instances': 2,
   'openshift_instances': 2},
  {'id': 2,
   'cloud_account_id': '699628107339',
   'user_id': 2,
   'type': 'aws',
   'arn': 'arn:aws:iam::699628107339:role/mock-arn',
   'creation_date': '2017-08-23T18:43:52.971743Z',
   'name': 'Second account',
   'images': 2,
   'instances': 2,
   'rhel_instances': 2,
   'openshift_instances': 1}]}
```

As can be seen on the API response, the number of RHEL instances for `My Account` should be 3 instead of 2 and the OCP should be 5 instead of 2. For the `Second Account` the numbers are correct because each image only had one instance running during the period.

## Steps to Reproduce

1. Add a cloud account
2. Add some usage for an image where more than one instance was run on the period
3. Fetch and the instance report, it should return the proper number of instances.
4. Fetch the accounts report, it won't the expected numbers.

## Expected Result

- Both instances and accounts reports should report the same numbers.

## Actual Result

- The numbers does not match.

## Additional context

- N/A

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Aug 23, 2018, 18:01

Using the account given on http://integrade-review-87-count-c-gqomck.review.gitlab.cloudigra.de/, some sample API responses...

For the root "accounts" page:

`/api/v1/report/accounts/?end=2018-08-24T00:00:00Z&start=2018-07-24T00:00:00Z`
```
{"cloud_account_overviews":[{"id":1,"cloud_account_id":"885089032249","user_id":2,"type":"aws","arn":"arn:aws:iam::885089032249:role/mock-arn","creation_date":"2017-08-23T18:43:46.784643Z","name":"My account","images":3,"instances":7,"rhel_instances":2,"openshift_instances":2},{"id":2,"cloud_account_id":"699628107339","user_id":2,"type":"aws","arn":"arn:aws:iam::699628107339:role/mock-arn","creation_date":"2017-08-23T18:43:52.971743Z","name":"Second account","images":2,"instances":2,"rhel_instances":2,"openshift_instances":1}]}
```

`/api/v1/report/instances/?end=2018-08-24T00:00:00Z&start=2018-07-24T00:00:00Z`
```
{"instances_seen_with_rhel":5,"instances_seen_with_openshift":6,"daily_usage":[{"date":"2018-07-24T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-25T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-26T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-27T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-28T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-29T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-30T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-31T00:00:00Z","rhel_instances":2,"openshift_instances":3,"rhel_runtime_seconds":108000.0,"openshift_runtime_seconds":201600.0},{"date":"2018-08-01T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":72000.0},{"date":"2018-08-02T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-03T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-04T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-05T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-06T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-07T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-08T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":68400.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-09T00:00:00Z","rhel_instances":2,"openshift_instances":2,"rhel_runtime_seconds":140400.0,"openshift_runtime_seconds":140400.0},{"date":"2018-08-10T00:00:00Z","rhel_instances":2,"openshift_instances":2,"rhel_runtime_seconds":104400.0,"openshift_runtime_seconds":172800.0},{"date":"2018-08-11T00:00:00Z","rhel_instances":2,"openshift_instances":3,"rhel_runtime_seconds":97200.0,"openshift_runtime_seconds":234000.0},{"date":"2018-08-12T00:00:00Z","rhel_instances":2,"openshift_instances":3,"rhel_runtime_seconds":162000.0,"openshift_runtime_seconds":259200.0},{"date":"2018-08-13T00:00:00Z","rhel_instances":1,"openshift_instances":3,"rhel_runtime_seconds":86400.0,"openshift_runtime_seconds":190800.0},{"date":"2018-08-14T00:00:00Z","rhel_instances":1,"openshift_instances":2,"rhel_runtime_seconds":32400.0,"openshift_runtime_seconds":118800.0},{"date":"2018-08-15T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-16T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-17T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-18T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-19T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-20T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-21T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-22T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-23T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":78602.383889}]}
```

For the detailed view on "My account":

`/api/v1/report/images/?account_id=1&end=2018-08-24T00:00:00Z&start=2018-07-24T00:00:00Z`
```
{"images":[{"id":2,"cloud_image_id":"500360200025","name":null,"status":"inspected","is_encrypted":false,"rhel":false,"rhel_detected":false,"rhel_challenged":false,"openshift":true,"openshift_detected":true,"openshift_challenged":false,"instances_seen":4,"runtime_seconds":2865146.604436},{"id":3,"cloud_image_id":"51226857049","name":null,"status":"inspected","is_encrypted":false,"rhel":true,"rhel_detected":true,"rhel_challenged":false,"openshift":false,"openshift_detected":false,"openshift_challenged":false,"instances_seen":2,"runtime_seconds":219600.0},{"id":4,"cloud_image_id":"83900912078","name":null,"status":"inspected","is_encrypted":false,"rhel":true,"rhel_detected":true,"rhel_challenged":false,"openshift":true,"openshift_detected":true,"openshift_challenged":false,"instances_seen":1,"runtime_seconds":72000.0}]}
```

`/api/v1/report/instances/?account_id=1&end=2018-08-24T00:00:00Z&start=2018-07-24T00:00:00Z`
```
{"instances_seen_with_rhel":3,"instances_seen_with_openshift":5,"daily_usage":[{"date":"2018-07-24T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-25T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-26T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-27T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-28T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-29T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-30T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-07-31T00:00:00Z","rhel_instances":2,"openshift_instances":3,"rhel_runtime_seconds":108000.0,"openshift_runtime_seconds":201600.0},{"date":"2018-08-01T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":72000.0},{"date":"2018-08-02T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-03T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-04T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-05T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-06T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-07T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-08T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":68400.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-09T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":86400.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-10T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":18000.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-11T00:00:00Z","rhel_instances":1,"openshift_instances":2,"rhel_runtime_seconds":10800.0,"openshift_runtime_seconds":147600.0},{"date":"2018-08-12T00:00:00Z","rhel_instances":0,"openshift_instances":2,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":172800.0},{"date":"2018-08-13T00:00:00Z","rhel_instances":0,"openshift_instances":2,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":104400.0},{"date":"2018-08-14T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-15T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-16T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-17T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-18T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-19T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-20T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-21T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-22T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-23T00:00:00Z","rhel_instances":0,"openshift_instances":1,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":78746.961351}]}
```

For the detailed view on "Second account":

`/api/v1/report/images/?account_id=2&end=2018-08-24T00:00:00Z&start=2018-07-24T00:00:00Z`
```
{"images":[{"id":3,"cloud_image_id":"51226857049","name":null,"status":"inspected","is_encrypted":false,"rhel":true,"rhel_detected":true,"rhel_challenged":false,"openshift":false,"openshift_detected":false,"openshift_challenged":false,"instances_seen":1,"runtime_seconds":75600.0},{"id":4,"cloud_image_id":"83900912078","name":null,"status":"inspected","is_encrypted":false,"rhel":true,"rhel_detected":true,"rhel_challenged":false,"openshift":true,"openshift_detected":true,"openshift_challenged":false,"instances_seen":1,"runtime_seconds":432000.0}]}
```

`/api/v1/report/instances/?account_id=2&end=2018-08-24T00:00:00Z&start=2018-07-24T00:00:00Z`
```
{"instances_seen_with_rhel":2,"instances_seen_with_openshift":1,"daily_usage":[{"date":"2018-07-24T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-25T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-26T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-27T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-28T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-29T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-30T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-07-31T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-01T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-02T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-03T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-04T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-05T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-06T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-07T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-08T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-09T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":54000.0,"openshift_runtime_seconds":54000.0},{"date":"2018-08-10T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":86400.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-11T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":86400.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-12T00:00:00Z","rhel_instances":2,"openshift_instances":1,"rhel_runtime_seconds":162000.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-13T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":86400.0,"openshift_runtime_seconds":86400.0},{"date":"2018-08-14T00:00:00Z","rhel_instances":1,"openshift_instances":1,"rhel_runtime_seconds":32400.0,"openshift_runtime_seconds":32400.0},{"date":"2018-08-15T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-16T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-17T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-18T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-19T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-20T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-21T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-22T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0},{"date":"2018-08-23T00:00:00Z","rhel_instances":0,"openshift_instances":0,"rhel_runtime_seconds":0.0,"openshift_runtime_seconds":0.0}]}
```

**Comment by katherine-black** [comment]
In GitLab by @erezende on Aug 23, 2018, 22:19

@infinitewarp checking how the reports are generated on the source code I found that for each instance a counter is incremented on the daily usage report https://gitlab.com/cloudigrade/cloudigrade/blob/master/cloudigrade/account/reports.py#L187-201.

The account summary report, on the other hand, does the rhel and openshift instance count based on the rhel list which is actually populated by rhel images https://gitlab.com/cloudigrade/cloudigrade/blob/master/cloudigrade/account/reports.py#L359-373. See on line 385 that rhel_instances is being set as total_rhel which is actually the total number of rhel images, see lines 365 and 366.

That said the logic on the `get_account_overview` function needs to be updated to actually count the instances and not the images.

I hope this help

**Comment by katherine-black** [comment]
In GitLab by @erezende on Aug 27, 2018, 10:03

After the fix for this being merged, I've redeployed Cloudigrade and got the expected numbers on the cards, see below:

![Screenshot_from_2018-08-27_10-53-14](/uploads/4698057245ad94d623f26b95a02d294f/Screenshot_from_2018-08-27_10-53-14.png)

I started with 2 RHEL + 1 OpenShift + 1 RHEL-OpenShift and the labels were properly displayed 3 RHEL and 2 OpenShift. After that I started adding more images and checking if the numbers were matching and they were.

With all that said I will close this and remove the needs verification label, this issue is now verified.
